django_cloud_deploy/tool_requirements.py

```diff
@@ -0,0 +1,270 @@
+        if not shutil.which('gcloud'):
+            download_link = 'https://cloud.google.com/sdk/install'
+            msg = ('Please download Google Cloud SDK'
+                   'from {}'.format(download_link))
+            raise MissingRequirementError(cls.NAME, msg)
+
+
+class Docker(Requirement):
+    NAME = 'Docker'
+
+    @classmethod
+    def check(cls):
+        """Checks if the requirement is installed.
+
+        Raises:
+            MissingRequirementError: If the requirement is not found.
+        """
+        if not shutil.which('docker'):
+            download_link = 'https://store.docker.com/'
+            msg = 'Please download Docker from {}'.format(download_link)
+            raise MissingRequirementError(cls.NAME, msg)
+
+        try:
+            command = ['docker', 'image', 'ls']
+            subprocess.check_call(command, stdout=subprocess.DEVNULL,
+                                  stderr=subprocess.DEVNULL)
+        except subprocess.CalledProcessError:
+            # TODO: Handle for multiple OS's
+            msg = ('Docker is installed but not correctly set up.'
+                   'Use the following command to fix it: '
+                   'sudo usermod -a -G docker $USER\n'
+                   'Then log out/log back in')
+            raise MissingRequirementError(cls.NAME, msg)
+
+
+class Kubectl(Requirement):
+    NAME = 'Kubectl'
+
+    @classmethod
+    def check(cls):
+        """Checks if the requirement is installed.
+
+        Raises:
+            MissingRequirementError: If the requirement is not found.
+        """
+        if not shutil.which('kubectl'):
+            msg = 'To install kubectl run: gcloud components install kubectl'
+            raise MissingRequirementError(cls.NAME, msg)
+
+
+class CloudSqlProxy(Requirement):
+    NAME = 'Cloud Sql Proxy'
+
+    @classmethod
+    def check(cls):
+        """Checks if the requirement is installed.
+
+        Raises:
+            MissingRequirementError: If the requirement is not found.
+        """
+        if not shutil.which('cloud_sql_proxy'):
+            dl_link = 'https://cloud.google.com/sql/docs/mysql/sql-proxy'
+            msg = 'Please download Cloud SQL Proxy from {}'.format(dl_link)
+            raise MissingRequirementError(cls.NAME, msg)
+
+    @classmethod
+    def handle(cls, console: io.IO):
+        """Attempts to install the requirement.
+
+        Raises:
+            UnableToAutomaticallyInstall: If the installation fails.
+            NotImplementedError: If we expect the user to manually install the
+            requirement.
+        """
+        can_download = cls._prompt_download_requirement(console)
+        if can_download:
+            cls._download_cloud_sql_proxy(console)
+        else:
+            raise UnableToAutomaticallyInstall
+
+    @classmethod
+    def _prompt_download_requirement(cls, console: io.IO) -> bool:
+        while True:
+            console.tell('{} is required. y/n to download requirements'
```

Comment:
I think that there is a prompt that does Y/n on a single line - maybe copy it for consistency?